src/rust/ensogl/src/display/shape/text/glyph/font.rs

```diff
@@ -267,19 +280,50 @@ impl FontRegistry {
     }
 
     /// Get handle one of loaded fonts.
-    pub fn get_render_info(&mut self, name:&str) -> Option<FontHandle> {
+    pub fn get_render_info(&mut self, name:&str) -> Option<Handle> {
         self.fonts.get_mut(name).cloned()
     }
 }
 
-impl Default for FontRegistry {
+impl Default for Registry {
     fn default() -> Self {
         Self::new()
     }
 }
 
 
 
+// ======================
+// === SharedRegistry ===
+// ======================
+
+/// Shared version of `Registry`.
+#[derive(Clone,CloneRef,Debug)]
+pub struct SharedRegistry {
+    model : Rc<RefCell<Registry>>
+}
+
+impl SharedRegistry {
+    /// Constructor.
+    #[allow(clippy::new_without_default)] // FIXME rename new to something more explicit
+    pub fn new() -> SharedRegistry {
+        let model = Rc::new(RefCell::new(Registry::new()));
+        Self {model}
+    }
+
+    /// Get render font info from loaded fonts, and if it does not exists, load data from one of
+    /// embedded fonts. Returns None if the name is missing in both loaded and embedded font list.
+    pub fn get_or_load_embedded_font(&self, name:&str) -> Option<Handle> {
+        self.model.borrow_mut().get_or_load_embedded_font(name)
+    }
+}
```

Comment:
I wonder if we could have just only one `Registry` which is internal mutable. Then everyone can decide if he want to keep it in Rc or not. And there will be only one structure to comprehend.

But it's up to you, do as you please.